### ISSUE TITLE
Repository finders results get mixed when run concurrently

### ISSUE THREAD

**Issue by frant-hartm** [spring-projects/spring-data-neo4j#382]
I am using 4.2.0.RC1 and ogm 2.1.0.

I have got following setup:
```World world1 = new World("world 1", 1);
worldRepository.save(world1, 0);

World world2 = new World("world 2", 2);
worldRepository.save(world2, 0);
```

and following queries running concurrently:
```
World world = worldRepository.findByName("world 1");
World world = worldRepository.findByName("world 2");
```

When run at almost the same time in concurrent threads one of the finders will return the other entity.

See attached test case.

**Comment by frant-hartm** [comment]
This is a regression. It was fixed by 9173958 and then reintroduced by f43d76d1

**Comment by mangrish** [comment]
Moved to: https://jira.spring.io/browse/DATAGRAPH-951
